### PR TITLE
fix(23804): unable to add QR code account

### LINKED ISSUE #23804: [Bug]: Hardware wallet QR connection unsuccessful
### Describe the bug

The connection via QR code on Hardware wallet is not proceeding past scanning the QR code. Functionality is working in the release version v11.13.1. 

### Expected behavior

The QR code should be scanned and you should be taken to the next screen of selecting the hardware wallet account you wish to add to MetaMask. 

### Screenshots/Recordings

https://github.com/MetaMask/metamask-extension/assets/120041701/da648f60-e044-4dd4-8e10-484f68dc4ee4

### Steps to reproduce

1. Load MetaMask
2. On your hardware wallet, fetch the QR code for MetaMask
3. Proceed to add hardware wallet from the account drop-down selector
4. Select QR-based and Continue
5. Scan the QR code from the hardware wallet 
6. Notice the QR code is scanned and nothing happens afterwards

### Error messages or log output

```shell
No console error
```

### Version

11.14.0

### Build type

None

### Browser

Chrome, Firefox

### Operating system

Windows, MacOS

### Hardware wallet

Keystone

### Additional context

_No response_

### Severity

Release-blocker

### PR DESCRIPTION
**Pull request by dawnseeker8**
## **Description**
The connection via QR code on Hardware wallet is not proceeding past scanning the QR code. Functionality is working in the release version v11.13.1.

After debuggin the code and discover that the issue is caused by Keyring Controller upgraded to 13.0.0. and `#addQRKeyring` function have bug to pass a empty accounts array as parameters to keyring, cause the Keystone MetamaskKeyring default properties all are undefined during deserialise() process.

This fix patch the `@metamask/keyring-controller` to remove the empty array passed to KeyStone MetamaskKeyring so that all default properies in keyring will remain the same.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23903?quickstart=1)

## **Related issues**

Fixes: #23804 

## **Manual testing steps**

1. On your hardware wallet, fetch the QR code for MetaMask
2. Proceed to add hardware wallet from the account drop-down selector
3. Select QR-based and Continue
4. Scan the QR code from the hardware wallet
5. user should be able to select QR code account and imported

### **Before**
https://private-user-images.githubusercontent.com/120041701/318063954-da648f60-e044-4dd4-8e10-484f68dc4ee4.mov?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2MzkzNTQsIm5iZiI6MTcxMjYzOTA1NCwicGF0aCI6Ii8xMjAwNDE3MDEvMzE4MDYzOTU0LWRhNjQ4ZjYwLWUwNDQtNGRkNC04ZTEwLTQ4NGY2OGRjNGVlNC5tb3Y_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwOVQwNTA0MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iM2QzMDllNTliMzNhNTc5YjVjMDJiY2M3MjZjYWU2NGExNWQ5OWY1MzI3NjExZDc3OWZmOGJhOTk5ODBhZDJjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.visSNnKwa_VELXLKh5PBUgbuMvBqOKMDH-oieICU_UI

### **After**

https://github.com/MetaMask/metamask-extension/assets/7315988/d4bd8765-499c-40b2-8429-117b58f61a9c

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -237,7 +237,10 @@
     "@metamask/network-controller@npm:^17.0.0": "patch:@metamask/network-controller@npm%3A18.0.1#~/.yarn/patches/@metamask-network-controller-npm-18.0.1-c4d0cfaecd.patch",
     "@metamask/network-controller@npm:^15.0.0": "patch:@metamask/network-controller@npm%3A18.0.1#~/.yarn/patches/@metamask-network-controller-npm-18.0.1-c4d0cfaecd.patch",
     "@metamask/network-controller@npm:^18.0.1": "patch:@metamask/network-controller@npm%3A18.0.1#~/.yarn/patches/@metamask-network-controller-npm-18.0.1-c4d0cfaecd.patch",
-    "@metamask/network-controller@npm:^17.2.1": "patch:@metamask/network-controller@npm%3A18.0.1#~/.yarn/patches/@metamask-network-controller-npm-18.0.1-c4d0cfaecd.patch"
+    "@metamask/network-controller@npm:^17.2.1": "patch:@metamask/network-controller@npm%3A18.0.1#~/.yarn/patches/@metamask-network-controller-npm-18.0.1-c4d0cfaecd.patch",
+    "@metamask/keyring-controller@npm:^12.2.0": "patch:@metamask/keyring-controller@npm%3A13.0.0#~/.yarn/patches/@metamask-keyring-controller-npm-13.0.0-d94816a680.patch",
```
Comment (danjm) [inline package.json:241]: Do we need to patch keyring controller v12.2.0 if the issue was only introduced in keyring controller v13.0.0?

File: package.json [hunk 2]
```diff
@@ -237,7 +237,10 @@
...
+    "@metamask/keyring-controller@npm:^14.0.1": "patch:@metamask/keyring-controller@npm%3A13.0.0#~/.yarn/patches/@metamask-keyring-controller-npm-13.0.0-d94816a680.patch",
```
Comment (danjm) [inline package.json:242]: Do we need a separate patch of v14.0.1, so that we don't downgrade it to v13?

File: .yarn/patches/@metamask-keyring-controller-npm-13.0.0-d94816a680.patch [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+--- a/dist/KeyringController.js
++++ b/dist/KeyringController.js
+@@ -1092,9 +1092,7 @@ _KeyringController_keyringBuilders = new WeakMap(), _KeyringController_keyrings
+ }, _KeyringController_addQRKeyring = function _KeyringController_addQRKeyring() {
+     return __awaiter(this, void 0, void 0, function* () {
+         // QRKeyring is not yet compatible with Keyring type from @metamask/utils
+-        const qrKeyring = (yield __classPrivateFieldGet(this, _KeyringController_instances, "m", _KeyringController_newKeyring).call(this, KeyringTypes.qr, {
+-            accounts: [],
+-        }));
++        const qrKeyring = (yield __classPrivateFieldGet(this, _KeyringController_instances, "m", _KeyringController_newKeyring).call(this, KeyringTypes.qr));
```
Comment (gantunesr) [inline .yarn/patches/@metamask-keyring-controller-npm-13.0.0-d94816a680.patch:12]: Can you add a comment to this patch explaining the reason for it?

**Review by gantunesr — APPROVED** [review]

File: .yarn/patches/@metamask-keyring-controller-npm-13.0.0-d94816a680.patch (hunk 3 above)
Comment (dawnseeker8) [reply .yarn/patches/@metamask-keyring-controller-npm-13.0.0-d94816a680.patch:12]: i will add it now.

**Review by gantunesr — CHANGES_REQUESTED** [review]
Dimiss approval

File: package.json (hunk 2 above)
Comment (dawnseeker8) [reply package.json:242]: same comments as above, this script was added by `yarn patch @metamask/keyring-controller@npm:13.0.0 ` and then `yarn patch-commit -s <directory>`. later i will try to see any impact on removing them.

Comment (dawnseeker8) [reply package.json]: i have removed all keyring-controller 12.2.0 and keyring-controller 14.0.1 patch in package.json, and i have done the testing, and it work quite well after removing those two entry in `package.json`.

File: package.json (hunk 1 above)
Comment (dawnseeker8) [reply package.json]: have removed that line in `package.json` and test locally with the change, and function still work quite well after removing the code.

**Comment by gantunesr** [comment]
@metamaskbot update-policies

File: package.json [hunk 4]
```diff
@@ -256,14 +257,14 @@
     "@lavamoat/lavadome-react": "0.0.17",
     "@lavamoat/snow": "^2.0.1",
     "@material-ui/core": "^4.11.0",
-    "@metamask-institutional/custody-controller": "^0.2.24",
-    "@metamask-institutional/custody-keyring": "^1.0.12",
-    "@metamask-institutional/extension": "^0.3.20",
-    "@metamask-institutional/institutional-features": "^1.2.15",
+    "@metamask-institutional/custody-controller": "^0.2.23",
```
Comment (Gudahtt) [inline package.json:260]: It looks like there are a number of MMI-related changes here - these packages are being downgraded. Is this intentional? It doesn't appear connected to the PR

**Review by benjisclowder — DISMISSED** [review]
QA passed.

File: package.json (hunk 4 above)
Comment (dawnseeker8) [reply package.json:260]: that is very weird, i look at the change, those 259 to 261 was added yesterday to `develop` branch , i will click `update branch` in this PR again to see any changes in `package.json`, if still have some library downgraded by `update branch` i will manually apply to `develop` branch version of those libraries.

**Review by benjisclowder — DISMISSED** [review]
QA passed

**Comment by Gudahtt** [comment]
@metamaskbot update-policies

**Review by benjisclowder — APPROVED** [review]
QA passed.

**Comment by danjm** [comment]
@SocketSecurity ignore-all

The new author warnings are safe. We have approved those authors in the past, and ther are github employees and/or long time contributors to the npm ecosystem.

The network access is necessary for make-fetch-happen to work, and that has been part of our dependency tree for a long time.

**Review by danjm — APPROVED** [review]

**Review by gantunesr — APPROVED** [review]
